### DISCUSSION TITLE
Build error + fix when using capnp/ez-rpc with bazel

### DISCUSSION THREAD

**Discussion by salil-enf** [capnproto/capnproto#1942] [General]
We use bazel to build and started using capnproto bazel rules directly (since last week). Earlier we were doing a configure_make step to pull in the repo. Using bazel-rules directly was must faster/easier however we ran into a build issue on the server side when we use ez-rpc. Including ez-rpc.h header leads to below build failure

```
bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/capnp/_virtual_includes/capnp-rpc/capnp/ez-rpc.h:25:10: fatal error: message.h: No such file or directory
   25 | #include "message.h"
      |          ^~~~~~~~~~~

```
The straight forward fix here of adding an include path did not seem to work. Because that started re-ordering include paths for all builds leading to problems like this below
```
In file included from bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/kj/_virtual_includes/kj/kj/string.h:24,
                 from bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/capnp/_virtual_includes/capnp/capnp/common.h:29,
                 from bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/capnp/_virtual_includes/capnp/capnp/endian.h:24,
                 from /usr/include/x86_64-linux-gnu/sys/types.h:176,
                 from /usr/include/stdlib.h:394,
                 from /usr/include/c++/10/cstdlib:75,
                 from /usr/include/c++/10/bits/stl_algo.h:59,
                 from /usr/include/c++/10/string:52
 ```

I tried this patch below and it seems to resolve the build issues for us.  Before I deploy, wanted to make sure I wasn't missing something obvious or if this is wrong because it could lead to future issues etc. Does this look okay?

```
@@ -0,0 +1,14 @@
+--- a/src/capnp/ez-rpc.h      2024-02-13 16:08:37.598140934 +0000
++++ b/src/capnp/ez-rpc.h      2024-02-13 16:08:49.558176863 +0000
+@@ -21,8 +21,8 @@
+
+ #pragma once
+
+-#include "rpc.h"
+-#include "message.h"
++#include "capnp/rpc.h"
++#include "capnp/message.h"
+
+ CAPNP_BEGIN_HEADER
```

**Comment by kentonv** [comment]
The correct change would be:

```diff
-#include "message.h"
+#include <capnp/message.h>
```

This is needed since `message.h` is in a separate bazel library, and bazel doesn't like relative includes cross-library. I guess it's surprising this doesn't cause `ez-rpc.c++` itself to fail to build.

(Actually, bazel prefers that all includes are absolute includes using quoted syntax. This is a weird Google-ism that I very much disagree with. In capnp we follow the rule that quoted includes are relative, `<>` includes are absolute.)

**Reply by salil-enf** [reply]
Thanks for following up on this. I am inclined to leave my patch as is for now since that is consistent with our include style.  Is this change planned from your side anytime soon?

**Reply by kentonv** [reply]
I'm happy to accept a PR fixing this, but I don't have the bandwidth to work on it myself.
